Title: SGX Attestation Fails Due to Oversized Endorsements from Intel endpoint (oe_verify_evidence returns OE_INVALID_PARAMETER)

We are encountering a failure in SGX attestation when using oe_verify_evidence() due to Intel's SGX endorsements exceeding the size currently supported by Open Enclave. The verification call fails with OE_INVALID_PARAMETER.

```
[/home/saeid/openenclave/host/sgx/sgxquoteprovider.c:oe_get_sgx_quote_verification_collateral:219]
2025-07-17T14:47:34+0000.005880Z [(H)INFO] tid(0x71d1ae213740) | root_ca_crl_size = 293
 [/home/saeid/openenclave/host/sgx/sgxquoteprovider.c:oe_get_sgx_quote_verification_collateral:234]
2025-07-17T14:47:34+0000.005883Z [(H)INFO] tid(0x71d1ae213740) | pck_crl_size = 6528
 [/home/saeid/openenclave/host/sgx/sgxquoteprovider.c:oe_get_sgx_quote_verification_collateral:254]
2025-07-17T14:47:34+0000.005886Z [(H)INFO] tid(0x71d1ae213740) | pck_crl_size = 1893
 [/home/saeid/openenclave/host/sgx/sgxquoteprovider.c:oe_get_sgx_quote_verification_collateral:270]
2025-07-17T14:47:34+0000.005890Z [(H)INFO] tid(0x71d1ae213740) | tcb_info_size = 6734
 [/home/saeid/openenclave/host/sgx/sgxquoteprovider.c:oe_get_sgx_quote_verification_collateral:285]
2025-07-17T14:47:34+0000.005893Z [(H)INFO] tid(0x71d1ae213740) | qe_identity_issuer_chain_size = 1893
 [/home/saeid/openenclave/host/sgx/sgxquoteprovider.c:oe_get_sgx_quote_verification_collateral:304]
2025-07-17T14:47:34+0000.005896Z [(H)INFO] tid(0x71d1ae213740) | qe_identity_size = 1381
 [/home/saeid/openenclave/host/sgx/sgxquoteprovider.c:oe_get_sgx_quote_verification_collateral:319]
2025-07-17T14:47:34+0000.005980Z [(E)INFO] tid(0x71d1ae213740) | enclave_b.signed:Enter call oe_create_sgx_endorsements
 [/home/saeid/openenclave/common/sgx/endorsements.c:oe_create_sgx_endorsements:39]
2025-07-17T14:47:34+0000.006012Z [(E)ERROR] tid(0x71d1ae213740) | enclave_b.signed:SGX endorsements are too large. Size is 20704 bytes (oe_result_t=OE_INVALID_PARAMETER) [/home/saeid/openenclave/common/sgx/endorsements.c:oe_create_sgx_endorsements:63]
2025-07-17T14:47:34+0000.006032Z [(E)INFO] tid(0x71d1ae213740) | enclave_b.signed:Exit call oe_create_sgx_endorsements: 3(OE_INVALID_PARAMETER)
 [/home/saeid/openenclave/common/sgx/endorsements.c:oe_create_sgx_endorsements:234]
2025-07-17T14:47:34+0000.006055Z [(E)ERROR] tid(0x71d1ae213740) | enclave_b.signed:Failed to create SGX endorsements. OE_INVALID_PARAMETER (oe_result_t=OE_INVALID_PARAMETER) [/home/saeid/openenclave/common/sgx/endorsements.c:oe_get_sgx_endorsements:413]
2025-07-17T14:47:34+0000.006090Z [(E)INFO] tid(0x71d1ae213740) | enclave_b.signed:Exit call oe_get_sgx_endorsements: 3(OE_INVALID_PARAMETER)
 [/home/saeid/openenclave/common/sgx/endorsements.c:oe_get_sgx_endorsements:425]
2025-07-17T14:47:34+0000.006109Z [(E)ERROR] tid(0x71d1ae213740) | enclave_b.signed::OE_INVALID_PARAMETER [/home/saeid/openenclave/common/sgx/verifier.c:oe_sgx_verify_evidence:882]
2025-07-17T14:47:34+0000.006126Z [(E)ERROR] tid(0x71d1ae213740) | enclave_b.signed::OE_INVALID_PARAMETER [/home/saeid/openenclave/common/attest_plugin.c:oe_verify_evidence:438]
Enclave2: ***../common/attestation.cpp(200): oe_verify_evidence failed (OE_INVALID_PARAMETER).

2025-07-17T14:47:34+0000.006180Z [(E)INFO] tid(0x71d1ae213740) | enclave_b.signed:attesters is NULL [/home/saeid/openenclave/enclave/sgx/attester.c:oe_attester_shutdown:462]
2025-07-17T14:47:34+0000.006198Z [(E)INFO] tid(0x71d1ae213740) | enclave_b.signed:free verifiers_length=4 plugins [/home/saeid/openenclave/common/sgx/verifier.c:oe_verifier_shutdown:1153]
Enclave2: ***../common/dispatcher.cpp(221): verify_evidence_and_set_public_key failed.

```
Impact:

This issue breaks remote attestation in any flow that depends on oe_verify_evidence() (or APIs relying on it), effectively halting production workloads when the enclave restarts and re-attestation is needed.

Environment:

    OE version: v0.19.11 (and reproducible on latest main)

    Platform: Intel SGX

    OS: Ubuntu 22.04

    Quoting Library: Latest QPL

    Endorsement size observed: 20704 bytes

Root Cause:

Intel has silently increased the size of SGX endorsements beyond what OE currently supports internally. The buffer size or validation logic in oe_create_sgx_endorsements() does not account for this, causing the operation to fail.

Suggested Fix:

Please increase the supported endorsement buffer size and update relevant checks in endorsements.c.

> [@Saeid2k](https://github.com/Saeid2k) Hi, could you share your step to reproduce the issue with latest main? Also, could you share the Intel endpoint configured with QPL? Thanks

@asvrada To reproduce the issue, simply use the Azure configuration from the Intel DCAP QPL ([link](https://github.com/intel/SGXDataCenterAttestationPrimitives/blob/main/QuoteGeneration/qcnl/linux/sgx_default_qcnl_azure.conf)), then run the OpenEnclave attestation sample with the following command:
`OE_LOG_LEVEL=INFO make runsgxremote
`
You will encounter the following error:
```
[/home/saeid/openenclave/host/sgx/sgxquoteprovider.c:oe_get_sgx_quote_verification_collateral:219]
2025-07-17T21:15:02+0000.551828Z [(H)INFO] tid(0x71bc06097740) | root_ca_crl_size = 293
 [/home/saeid/openenclave/host/sgx/sgxquoteprovider.c:oe_get_sgx_quote_verification_collateral:234]
2025-07-17T21:15:02+0000.551833Z [(H)INFO] tid(0x71bc06097740) | pck_crl_size = 6527
 [/home/saeid/openenclave/host/sgx/sgxquoteprovider.c:oe_get_sgx_quote_verification_collateral:254]
2025-07-17T21:15:02+0000.551836Z [(H)INFO] tid(0x71bc06097740) | pck_crl_size = 1893
 [/home/saeid/openenclave/host/sgx/sgxquoteprovider.c:oe_get_sgx_quote_verification_collateral:270]
2025-07-17T21:15:02+0000.551839Z [(H)INFO] tid(0x71bc06097740) | tcb_info_size = 6734
 [/home/saeid/openenclave/host/sgx/sgxquoteprovider.c:oe_get_sgx_quote_verification_collateral:285]
2025-07-17T21:15:02+0000.551841Z [(H)INFO] tid(0x71bc06097740) | qe_identity_issuer_chain_size = 1893
 [/home/saeid/openenclave/host/sgx/sgxquoteprovider.c:oe_get_sgx_quote_verification_collateral:304]
2025-07-17T21:15:02+0000.551843Z [(H)INFO] tid(0x71bc06097740) | qe_identity_size = 1381
 [/home/saeid/openenclave/host/sgx/sgxquoteprovider.c:oe_get_sgx_quote_verification_collateral:319]
Enclave2: ***../common/attestation.cpp(201): oe_verify_evidence failed (OE_INVALID_PARAMETER).

Enclave2: ***../common/dispatcher.cpp(221): verify_evidence_and_set_public_key failed.
Host: verify_evidence_and_set_public_key failed. OE_OK
Host: attestation failed with 1
Host: Terminating enclaves
Enclave1: ***../common/crypto.cpp(89): openssl 3 cleaned up.
Host: Enclave successfully terminated.
Enclave2: ***../common/crypto.cpp(89): openssl 3 cleaned up.
Host: Enclave successfully terminated.
Host:  failed 
2025-07-17T21:15:02+0000.574426Z [(H)INFO] tid(0x71bc06097740) | _unload_quote_provider libdcap_quoteprov.so
 [/home/saeid/openenclave/host/sgx/linux/sgxquoteproviderloader.c:_unload_quote_provider:15]
2025-07-17T21:15:02+0000.574616Z [(H)INFO] tid(0x71bc06097740) | _unload_quote_ex_library() libsgx_quote_ex.so.1
 [/home/saeid/openenclave/host/sgx/linux/sgxquoteexloader.c:_unload_quote_ex_library:22]
make: *** [Makefile:27: runsgxremote] Error 1
saeid@saeid-dev-ubuntu-22:~/openenclave/samples/attestation$ 

```
I initially encountered this issue with the latest OE attestation sample. To debug it further, I used an older version of the OE attestation sample, which provides more detailed logs. The message I shared with you earlier comes from that older version, as it offers more granular logging compared to what is currently available in the OE main branch.

BDW, I’ve seen your fix PR and can confirm that your changes should resolve the issue. Please go ahead and merge the PR.

--- a/include/openenclave/bits/attestation.h
+++ b/include/openenclave/bits/attestation.h
@@ -14,7 +14,7 @@
 #include <openenclave/bits/types.h>
 
 /*! Limit the size of the endorsements */
-#define OE_ATTESTATION_ENDORSEMENT_MAX_SIZE (20 * 1024)
+#define OE_ATTESTATION_ENDORSEMENT_MAX_SIZE (200 * 1024)
 
 /*! Endorsement structure version */
 #define OE_ATTESTATION_ENDORSEMENT_VERSION (1)

--- a/tests/CMakeLists.txt
+++ b/tests/CMakeLists.txt
@@ -172,8 +172,9 @@ if (UNIX
     add_subdirectory(SampleApp)
     add_subdirectory(SampleAppCRT)
     add_subdirectory(sealKey)
-    add_subdirectory(secure_verify)
-    add_subdirectory(intel_qve_thread_test)
+    # Diabled due to TDX QvE verification error
+    # add_subdirectory(secure_verify)
+    # add_subdirectory(intel_qve_thread_test)
     add_subdirectory(sim-mode)
     add_subdirectory(stdc)
     add_subdirectory(stdcxx)

